Crash in TestSerf_ReapHandler_Shutdown test

```
panic: Fail in goroutine after TestSerf_ReapHandler_Shutdown has completed

goroutine 1033 [running]:
panic(0x432320, 0xc42028c4c0)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
testing.(*common).Fail(0xc4201ca180)
        /usr/local/go/src/testing/testing.go:412 +0x11f
testing.(*common).FailNow(0xc4201ca180)
        /usr/local/go/src/testing/testing.go:431 +0x2b
testing.(*common).Fatalf(0xc4201ca180, 0x4cbf19, 0x7, 0x0, 0x0, 0x0)
        /usr/local/go/src/testing/testing.go:496 +0x83
github.com/hashicorp/serf/serf.TestSerf_ReapHandler_Shutdown.func1(0xc4202da480, 0xc4201ca180)
        /Users/james/projects/_go/src/github.com/hashicorp/serf/serf/serf_test.go:841 +0x78
created by github.com/hashicorp/serf/serf.TestSerf_ReapHandler_Shutdown
        /Users/james/projects/_go/src/github.com/hashicorp/serf/serf/serf_test.go:842 +0x7b
FAIL    github.com/hashicorp/serf/serf  5.860s
ok      github.com/hashicorp/serf/testutil      0.022s
make: *** [test] Error 1
```

--- a/serf/serf_test.go
+++ b/serf/serf_test.go
@@ -835,12 +835,19 @@ func TestSerf_ReapHandler_Shutdown(t *testing.T) {
 		t.Fatalf("err: %s", err)
 	}
 
+	// Make sure the reap handler exits on shutdown.
+	doneCh := make(chan struct{})
 	go func() {
-		s.Shutdown()
-		time.Sleep(time.Millisecond)
-		t.Fatalf("timeout")
+		s.handleReap()
+		close(doneCh)
 	}()
-	s.handleReap()
+
+	s.Shutdown()
+	select {
+	case <-doneCh:
+	case <-time.After(1 * time.Second):
+		t.Fatalf("timeout")
+	}
 }
 
 func TestSerf_ReapHandler(t *testing.T) {

